### PR TITLE
MAT-5432 - useContext for Library

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5432](https://jira.cms.gov/browse/MAT-5432)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java [hunk 1]
```diff
@@ -74,13 +76,32 @@ public Library convertToFhirLibrary(CqlLibrary cqlLibrary) {
...
+    useContext.setCode(code);
+    CodeableConcept valueCodeableConcept = new CodeableConcept();
+    Coding coding = new Coding();
+    coding.setSystem("http://hl7.org/fhir/us/cqfmeasures/CodeSystem/quality-programs");
+    coding.setCode(programUseContext.getCode());
+    coding.setDisplay(programUseContext.getDisplay());
+    valueCodeableConcept.setCoding(List.of(coding));
+    useContext.setValue(valueCodeableConcept);
+    return useContext;
+    //
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java]: Remove or populate empty comment.

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java [hunk 2]
```diff
@@ -74,13 +76,31 @@ public Library convertToFhirLibrary(CqlLibrary cqlLibrary) {
...
+    }
     // TODO: probably have to revisit this. Human Readable feature is not yet ready
     // result.setText(findHumanReadable(lib.getMeasureId()));
     return library;
   }
 
+  public UsageContext convertUseContext(ProgramUseContext programUseContext) {
+    UsageContext useContext = new UsageContext();
+    Coding code = new Coding();
+    code.setSystem("http://terminology.hl7.org/CodeSystem/usage-context-type");
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java]: you can probably move this url to `UriConstants`

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java [hunk 3]
```diff
@@ -74,13 +76,31 @@ public Library convertToFhirLibrary(CqlLibrary cqlLibrary) {
...
+    code.setCode("program");
+    useContext.setCode(code);
+    CodeableConcept valueCodeableConcept = new CodeableConcept();
+    Coding coding = new Coding();
+    coding.setSystem("http://hl7.org/fhir/us/cqfmeasures/CodeSystem/quality-programs");
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java]: and this one

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java [hunk 4]
```diff
@@ -74,13 +76,31 @@ public Library convertToFhirLibrary(CqlLibrary cqlLibrary) {
...
+    coding.setCode(programUseContext.getCode());
+    coding.setDisplay(programUseContext.getDisplay());
+    valueCodeableConcept.setCoding(List.of(coding));
+    useContext.setValue(valueCodeableConcept);
+    return useContext;
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java:101]: this is not going make any difference. just for your information, most of the setters methods in FHIR models return instance of the same model(or `this`), that let's you method chaining. e.g.

```
public UsageContext convertUseContext(ProgramUseContext programUseContext) {
    Coding code =
        new Coding()
            .setSystem("http://terminology.hl7.org/CodeSystem/usage-context-type")
            .setCode("program");

    Coding coding =
        new Coding()
            .setSystem("http://hl7.org/fhir/us/cqfmeasures/CodeSystem/quality-programs")
            .setCode(programUseContext.getCode())
            .setDisplay(programUseContext.getDisplay());
    CodeableConcept valueCodeableConcept = new CodeableConcept().setCoding(List.of(coding));

    return new UsageContext().setCode(code).setValue(valueCodeableConcept);
  }
```

Comment (ethankaplan) [reply src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java:101]: I didn't know this, thank you!

**Review by adongare — APPROVED** [review]

**Pull request merged**